app/services/deadline_calculator/business_days.rb

```diff
@@ -32,6 +32,21 @@ def business_unit_deadline_for_date(date)
       days_after_day_one.business_days.after(date.to_date)
     end
 
+    def extension_deadline(time_limit)
+      days_after_day_one = (time_limit + kase.correspondence_type.external_time_limit) - 1
+      days_after_day_one.business_days.after(start_date(kase.received_date))
+    end
+    
+    def max_allowed_deadline_date(time_limit=nil)
+      time_limit ||= (kase.correspondence_type.extension_time_limit || 0)
+      days_after_day_one = (time_limit + kase.correspondence_type.external_time_limit) - 1
+      days_after_day_one.business_days.after(start_date(kase.received_date))
+    end
+
+    def time_units_desc_for_deadline(plural=false)
+      plural ? "business days".freeze : "business day".freeze
+    end
```

Comment:
assuming the argument is the number of days
```suggestion
    def time_units_desc_for_deadline(time_limit)
      "business #{'day'.pluralize(time_limit)}".freeze
    end
```

Reply:
thank you for the suggestion!